pages/home/vercelai/use-arcade-tools.mdx

```diff
@@ -1,9 +1,153 @@
+import { Steps, Tabs } from "nextra/components";
 
-TODO
+## Use Arcade with Vercel AI SDK
 
+The [Vercel AI SDK](https://sdk.vercel.ai/) is an open-source library that simplifies the process of building AI-powered applications. It provides a consistent interface for working with various AI providers and includes several useful features:
+
+- Streaming AI responses for real-time interactions
+- Framework-agnostic support for React, Next.js, Vue, Nuxt, and SvelteKit.
+- Easy switching between AI providers with a single line of code
+
+Let's supercharge your Vercel AI SDK applications with Arcade's tools. You'll get instant access to production-ready tools for working with Google, Slack, GitHub, LinkedIn, and many other popular services, all with built-in authentication. Browse our [complete toolkit catalog](https://docs.arcade.dev/toolkits) to discover what you can build.
+
+In this guide, we'll show you how to use Arcade's Google toolkit to create an AI agent that can read and summarize emails. You can find the complete code in our [GitHub repository](https://github.com/ArcadeAI/arcade-ai/tree/main/examples/ai-sdk) or follow along below.
+
+## Getting started
+
+
+<Steps>
+
+### Install the required dependencies
+
+
+We'll use the `@ai-sdk/openai` package in this example, but you can use any AI provider supported by the Vercel AI SDK. See the [full list of providers](https://ai-sdk.dev/docs/foundations/providers-and-models#ai-sdk-providers).
+
+<Tabs items={["pnpm", "npm", "yarn"]}>
+
+<Tabs.Tab>
+
+```bash
+pnpm add ai @ai-sdk/openai @arcadeai/arcadejs
+```
+
+</Tabs.Tab>
+
+<Tabs.Tab>
+
+```bash
+npm install 
+```
+
+</Tabs.Tab>
+
+<Tabs.Tab>
+
+```bash
+yarn install 
+```
+
+</Tabs.Tab>
+
+</Tabs>
+
+### Get your API keys and set up environment variables
+
+You'll need two API keys:
+- **OpenAI API key** from [OpenAI dashboard](https://platform.openai.com/api-keys)
+- **Arcade API key** from our [Getting your API key guide](https://docs.arcade.dev/home/api-keys)
+
+Add them to your environment:
+
+```bash
+OPENAI_API_KEY=<YOUR_OPENAI_API_KEY>
+ARCADE_API_KEY=<YOUR_ARCADE_API_KEY>
+```
+
+
+### Import Libraries and Initialize Arcade
+
+```ts
+import { openai } from "@ai-sdk/openai"
+import { generateText } from "ai"
+import { Arcade } from "@arcadeai/arcadejs"
+import { toZodToolSet, executeOrAuthorizeZodTool } from "@arcadeai/arcadejs/lib"
+
+const arcade = new Arcade()
+```
+
+### Get tools from Arcade's Google toolkit and convert them to Zod
+
+```ts
+// Get Arcade's google toolkit
+const googleToolkit = await arcade.tools.list({ toolkit: "google", limit: 30 })
+const googleTools = toZodToolSet({
+    tools: googleToolkit.items,
+    client: arcade,
+    userId: "<YOUR_USER_ID>", // Your app's internal ID for the user (an email, UUID, etc). It's used internally to identify your user in Arcade
+    executeFactory: executeOrAuthorizeZodTool, // Checks if tool is authorized and executes it, or returns authorization URL if needed
+})
```

Comment:
From the comments in https://github.com/ArcadeAI/arcade-ai/pull/396 - I think we need some commentary about /why/ transforming things to Zod is required.  It's the common object-mapper that everyone seems to be using... why is that good?